pandas/core/indexes/multi.py

```diff
@@ -2628,13 +2628,25 @@ def _partial_tup_index(self, tup, side="left"):
                     loc -= 1
                 return start + section.searchsorted(loc, side=side)
 
-            idx = lev.get_loc(lab)
+            idx = self._get_loc_single_index_wrapper(lev, lab)
             if k < n - 1:
                 end = start + section.searchsorted(idx, side="right")
                 start = start + section.searchsorted(idx, side="left")
             else:
                 return start + section.searchsorted(idx, side=side)
 
+    def _get_loc_single_index_wrapper(self, level_index, key):
+        """
+        single index get_loc wrapper function for multi index.
+        Depending on index type, Handling NA value is different using get_loc.
+        But in multi index, NA values is denoted as missing by -1.
+        If key is NA value, location of index unify as -1.
+        """
+        if not is_list_like(key) and isna(key):
+            return -1
+        else:
```

Comment:
rather than having changing MultiIndex like this, can you just update .get_loc on Index? e.g. it correctly returns -1 for a null input?

Reply:
@jreback 
Well, That is more than this issue. For example,
```
In [2]: idx = pd.Index(['a','b',None,'c','d']) 
   ...: idx.get_loc(None)                                                       
Out[2]: 2
```
Like this case, some Index `.get_loc` can find NA value. I thought it meant NA value doesn't consider as missing. But if return -1 for a null input, then it regards as missing. From then, this PR get off topic. this PR deals with Multiindex's missing and single index missing both. I think for single index separate issue.